### ISSUE TITLE
visualization？

### ISSUE THREAD

**Issue by milkyash628** [valeoai/WaffleIron#16]

**Comment by gpuy** [comment]
I have no clean code ready to share for visualisation. 

If this is about visualizing the results of the segmentation, the easiest is to use the scripts `eval_nuscenes.py` or `eval_kitti.py` to save the labels on disk. The results can be visualized with, e.g., Open3D, CloudCompare, etc.

For the visualisation of the feature maps, I directly saved the features on disk at this [location](https://github.com/valeoai/WaffleIron/blob/master/waffleiron/backbone.py#L176). After removing the batch dimension, we should get an array of size C x N x M. Most of the C-dimensional features are zeros (non-occupied cells). For the occupied cells, I extracted the (non-zero) features, used t-SNE to reduce the feature size C to 3. PCA should work too. These 3 values are used as RGB color after rescaling to the range [0...1]. The non-occupied cells on their side keep their zero value (black). I then saved the 3 x N x M array as an image.
